DRY out Phalanx

athornton opened the pull request:
(cachemachine first)

jonathansick commented on science-platform/templates/alert-stream-broker-application.yaml:
```diff
@@ -24,6 +24,14 @@ spec:
+          value: {{ .Values.fqdn | quote }}
+        - name: "global.baseUrl"
+          value: "https://{{ .Values.fqdn }}"
+        - name: "global.vaultSecretsPath"
```
I think I called this `global.vaultSecretsPathPrefix` in my apps; I don't know if you want to do the same?

rra replied:
I'd be good with that, but we should be consistent one way or the other across all applications, so we should probably fix the existing ones in one direction or the other.

athornton replied:
Y'all figure out which way it should go, and I'm happy to make it consistent.  I prefer the shorter version, myself.

rra reviewed (changes requested):
I'd rather we not add the global settings to every application without regard to whether they use them. Third-party applications mostly can't do anything useful with those settings, and will need other settings injected.

I'd rather we add the Helm parameters at the same time that we port a chart so that each change is self-contained.

rra commented on services/datalinker/Chart.yaml:
```diff
@@ -1,10 +1,8 @@
-    repository: https://lsst-sqre.github.io/charts/
+type: application
+version: 0.1.6
+maintainers:
+  - name: cbanek
```
Delete maintainers when you move charts over.

athornton replied:
OK.

rra commented on services/datalinker/templates/NOTES.txt:
```diff
@@ -0,0 +1,22 @@
+1. Get the application URL by running these commands:
```
Delete this file; it's only used for manual Helm installations and doesn't do anything useful in an Argo CD environment.

athornton replied:
done

rra commented on services/datalinker/templates/deployment.yaml:
```diff
@@ -0,0 +1,67 @@
+        {{- include "datalinker.selectorLabels" . | nindent 8 }}
+    spec:
+      automountServiceAccountToken: false
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
```
This should no longer be a template or in `values.yaml` and should instead be hard-coded with the value of `pull-secret` as created by the chart.

rra commented on services/datalinker/templates/ingress.yaml:
```diff
@@ -0,0 +1,35 @@
+{{- if .Values.ingress.enabled -}}
```
Remove this conditional and make the ingress unconditional.

rra commented on services/datalinker/templates/ingress.yaml:
```diff
@@ -0,0 +1,35 @@
+    {{- if .Values.ingress.gafaelfawrAuthQuery -}}
+    nginx.ingress.kubernetes.io/auth-method: GET
+    nginx.ingress.kubernetes.io/auth-response-headers: X-Auth-Request-User,X-Auth-Request-Email,X-Auth-Request-Token
+    nginx.ingress.kubernetes.io/auth-signin: "https://{{ .Values.global.host }}/login"
+    nginx.ingress.kubernetes.io/auth-url: "https://{{ .Values.global.host }}/auth?{{ .Values.ingress.gafaelfawrAuthQuery }}"
```
Use `.Values.global.baseUrl` instead.

rra commented on services/datalinker/templates/ingress.yaml:
```diff
@@ -0,0 +1,35 @@
+  name: {{ $fullName }}
+  labels:
+    {{- include "datalinker.labels" . | nindent 4 }}
+  {{- with .Values.ingress.annotations }}
```
Nit: Move this line down to where the annotations are actually used to make it match the pattern of other services.

athornton replied:
done

rra commented on services/datalinker/templates/ingress.yaml:
```diff
@@ -0,0 +1,35 @@
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  {{- if .Values.ingress.className }}
+  ingressClassName: {{ .Values.ingress.className }}
+  {{- end }}
```
This shouldn't be configurable any more.

rra commented on services/datalinker/templates/ingress.yaml:
```diff
@@ -0,0 +1,35 @@
+  ingressClassName: {{ .Values.ingress.className }}
+  {{- end }}
+  rules:
+    - host: {{ required "ingress.host must be set" .Values.global.host | quote }}
```
Error message is now wrong (should be `global.host`).

athornton replied:
done

rra commented on services/datalinker/templates/ingress.yaml:
```diff
@@ -0,0 +1,35 @@
+    - host: {{ required "ingress.host must be set" .Values.global.host | quote }}
+      http:
+        paths:
+          - path: {{ .Values.ingress.path }}
+            pathType: {{ default "Prefix" .Values.ingress.pathType }}
```
These shouldn't be configurable.

rra commented on services/datalinker/templates/ingress.yaml:
```diff
@@ -0,0 +1,35 @@
+              service:
+                name: {{ $fullName }}
+                port:
+                  number: {{ .Values.service.port }}
```
This shouldn't be configurable; just hard-code the correct value.

athornton replied:
done

rra commented on services/datalinker/templates/service.yaml:
```diff
@@ -0,0 +1,15 @@
+  labels:
+    {{- include "datalinker.labels" . | nindent 4 }}
+spec:
+  type: {{ .Values.service.type }}
+  ports:
+    - port: {{ .Values.service.port }}
```
Hard-code these values now.

athornton replied:
done

rra commented on services/datalinker/templates/vault-pull-secret.yaml:
```diff
@@ -0,0 +1,9 @@
+apiVersion: ricoberger.de/v1alpha1
```
This file should be named `vault-secrets.yaml` to match the other applications.

athornton replied:
done

rra commented on services/datalinker/values.yaml:
```diff
@@ -0,0 +1,81 @@
+  # -- Overrides the image tag whose default is the chart appVersion.
+  tag: ""
+
+# -- Secret names to use for all Docker pulls
+imagePullSecrets: 
+  - name: "pull-secret"
```
Delete.

athornton replied:
done

athornton commented on services/datalinker/values-idfdev.yaml:
```diff
@@ -1,8 +1 @@
-pull-secret:
-  enabled: true
-  path: secret/k8s_operator/data-dev.lsst.cloud/pull-secret
+# idfdev
```
`git push` wasn't pushing zero-length files for me.  Is that a git setting I have somewhere?

athornton replied:
Musta been PEBCAK

athornton commented:
> I'd rather we not add the global settings to every application without regard to whether they use them. Third-party applications mostly can't do anything useful with those settings, and will need other settings injected.
> 
> I'd rather we add the Helm parameters at the same time that we port a chart so that each change is self-contained.

OK, backing out the ones I haven't done yet.

athornton commented on services/datalinker/values.yaml:
```diff
@@ -0,0 +1,81 @@
+nameOverride: ""
+
+# -- Override the full name for resources (includes the release name)
+fullnameOverride: ""
```
done

athornton commented on services/datalinker/values.yaml:
```diff
@@ -0,0 +1,81 @@
+  pathType: ImplementationSpecific
+
+  # -- URL path to dispatch to the datalinker deployment pod
+  path: "/api/datalink"
```
done

athornton commented:
I think it's ready to look at again.  I believe I did all the stuff Russ asked, consistently (I hope) across repositories.

The sasquatch/strimzi-registry-operator interaction is still kind of opaque to me, and nublado2 would need some code changes if we wanted to REALLY get rid of the pull-and-butler-secret config in the environment values files.

I also removed pull secrets from those repositories (mobu, argocd, semaphore, squareone) where we do not need them and we are not pulling from Docker Hub or GAR or T&S Nexus.

athornton commented:
It may be easier to look at this commit-by-commit, which largely tracks individual applications' migration.

afausti reviewed (changes requested):
Note that `strimzi-registry-operator` is an example of a service that is shared among other phalanx services, and per RFC-830 it should be located in phalanx/charts/ and symlinked from the services that use it. I think this should be done in a separate PR along with the changes in sasquatch and the alert-stream-broker.

athornton commented:
I think I did what Angelo requested; I am not currently testing in a place with strimzi-registry-operator enabled, though.

rra commented on services/exposurelog/values.yaml:
```diff
@@ -1,6 +1,8 @@
 # Default values for exposurelog.
```
Needs helm-docs comments added for all the settings.

athornton replied:
Done

rra commented on services/moneypenny/README.md:
```diff
@@ -0,0 +1,38 @@
+# moneypenny
```
Needs `README.md.gotmpl`

rra commented on services/moneypenny/templates/ingress.yaml:
```diff
@@ -0,0 +1,36 @@
+                name: {{ include "moneypenny.fullname" . }}
+                port:
+                  number: 8080
+  {{- if .Values.ingress.tls }}
+  tls:
+    {{- range .Values.ingress.tls }}
+    - hosts:
+      {{- range .hosts }}
+        - {{ . | quote }}
+      {{- end }}
+      secretName: {{ .secretName }}
+    {{- end }}
+  {{- end }}
```
Delete all the TLS stuff.

athornton replied:
Done

rra commented on services/moneypenny/templates/serviceaccount.yaml:
```diff
@@ -0,0 +1,8 @@
+  name: {{ include "moneypenny.serviceAccountName" . }}
+  labels:
+    {{- include "moneypenny.labels" . | nindent 4 }}
+imagePullSecrets:
+  - name: "pull-secret"
```
Similarly pointless here. This is already set on the deployments.

athornton replied:
imagePullSecrets removed

rra commented on services/narrativelog/README.md:
```diff
@@ -1,6 +1,38 @@
 # narrativelog
```
Needs `README.md.gotmpl`

rra commented on services/narrativelog/values.yaml:
```diff
@@ -1,6 +1,8 @@
 # Default values for narrativelog.
```
Needs helm-docs comments for all the settings.

athornton replied:
Done.

rra commented on services/obstap/README.md:
```diff
@@ -0,0 +1,47 @@
+# cadc-tap-postgres
```
Needs `README.md.gotmpl`

rra commented on services/plot-navigator/README.md:
```diff
@@ -0,0 +1,18 @@
+# plot-navigator
```
Needs `README.md.gotmpl`

rra commented on services/plot-navigator/templates/deployment.yaml:
```diff
@@ -0,0 +1,66 @@
+---
```
Stray document separator.

athornton replied:
done

rra commented on services/plot-navigator/templates/deployment.yaml:
```diff
@@ -0,0 +1,66 @@
+  replicas: 1
+  selector:
+    matchLabels:
+      app: plot-navigator
+  template:
+    metadata:
+      labels:
+        app: plot-navigator
```
Should use the standard selector labels.

athornton replied:
Done

rra commented on services/plot-navigator/templates/ingress.yaml:
```diff
@@ -0,0 +1,30 @@
+---
```
Stray document separator

rra commented on services/plot-navigator/templates/service.yaml:
```diff
@@ -0,0 +1,13 @@
+  labels:
+    {{- include "plot-navigator.labels" . | nindent 4 }}
+spec:
+  selector:
+    app: plot-navigator
```
Should use the standard selector labels

rra commented on services/plot-navigator/templates/vault-secrets.yaml:
```diff
@@ -0,0 +1,20 @@
+---
```
Stray document separator

rra commented on services/postgres/README.md:
```diff
@@ -0,0 +1,21 @@
+# postgres
```
Needs `README.md.gotmpl`

rra commented on services/postgres/templates/_helpers.tpl:
```diff
@@ -0,0 +1,56 @@
+{{- else -}}
+    {{ default "default" .Values.serviceAccount.name }}
+{{- end -}}
+{{- end -}}
```
Missing the standard selectorLabels template

athornton replied:
Done

rra commented on services/postgres/templates/deployment.yaml:
```diff
@@ -0,0 +1,95 @@
+    app: {{ template "postgres.fullname" . }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      name: {{ template "postgres.fullname" . }}
+  template:
+    metadata:
+      labels:
+        name: {{ template "postgres.fullname" . }}
```
Should use the standard selector labels

athornton replied:
Done

rra commented on services/postgres/templates/deployment.yaml:
```diff
@@ -0,0 +1,95 @@
+      containers:
+        - name: {{ template "postgres.fullname" . }}
+          imagePullPolicy: "Always"
+          image: {{ .Values.image.repository }}:{{ .Values.image.tag }}
```
Should default to the appVersion

athornton replied:
Done

rra commented on services/postgres/templates/physpvc.yaml:
```diff
@@ -0,0 +1,16 @@
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
```
Missing standard labels

athornton replied:
Done

rra commented on services/postgres/templates/physpvc.yaml:
```diff
@@ -0,0 +1,16 @@
+    requests:
+      storage: {{.Values.postgres_volume_size}}
+  storageClassName: {{.Values.postgres_storage_class}}
+{{ if .Values.volume_name }}
+  volumeName: {{ .Values.volume_name }}
+{{ end }}
+
+
```
Stray blank lines and should use `-` to delete unwanted whitespace for the `if` and `end` macros.

athornton replied:
Done

rra commented on services/postgres/templates/service.yaml:
```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+kind: Service
+metadata:
```
Missing standard labels

rra commented on services/postgres/templates/service.yaml:
```diff
@@ -0,0 +1,10 @@
+  ports:
+    - name: postgres
+      port: 5432
+  selector:
+    name: {{ template "postgres.fullname" . }}
```
Should use standard selector labels

rra commented on services/postgres/templates/storageclass.yaml:
```diff
@@ -0,0 +1,9 @@
+{{- if eq .Values.postgres_storage_class "fast" }}
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: fast
+provisioner: kubernetes.io/gce-pd
+parameters:
+  type: pd-ssd
+{{- end }}
```
Do we care about this any more? If we want a good backend, wouldn't we just use Cloud SQL?

athornton replied:
That's probably the right thing, but our current postgres implementations are all using "fast" and putting the postgres volume on SSD currently.

rra commented on services/postgres/values-base.yaml:
```diff
@@ -1,25 +1,16 @@
-  enabled: true
-  path: secret/k8s_operator/base-lsp.lsst.codes/pull-secret
+jupyterhub_db:
+  user: 'jovyan'
```
Here and in the other values files, these are using single quotes, but we use double quotes everywhere else. While we're doing this sort of big refactoring, we should probably be consistent.

athornton replied:
Done

rra commented on services/postgres/values.yaml:
```diff
@@ -0,0 +1,18 @@
+# Default values for fileserver.
```
Needs helm-docs comments. The standard for Helm parameters is studlyCaps, not snake_case. Looks like most of the parameters aren't documented and probably should be.

athornton replied:
Done (and converted to studlyCaps)

rra commented on services/sherlock/README.md:
```diff
@@ -0,0 +1,33 @@
+# sherlock
```
Needs `README.md.gotmpl`

rra commented on services/sherlock/values.yaml:
```diff
@@ -0,0 +1,67 @@
+  name: ""
+
+# -- URL to push status to via HTTP PUTs.
+publish_url: ""
```
Should use studlyCaps here as well.

athornton replied:
Done.

rra commented on services/strimzi-registry-operator/Chart.yaml:
```diff
@@ -1,7 +1,5 @@
 apiVersion: v2
```
I'm confused; strimzi-registry-operator was added as a shared chart earlier. It shouldn't be both a service and a shared chart; it should be one or the other, no?

rra commented on services/strimzi-registry-operator/values.yaml:
```diff
@@ -0,0 +1,11 @@
+image:
```
Missing helm-docs comments for several parameters.

rra commented on services/tap/README.md:
```diff
@@ -0,0 +1,54 @@
+# cadc-tap
```
Needs `README.md.gotmpl`

rra reviewed (commented):
I didn't check all the `values.yaml` files, but I think some of the ones you moved over from charts are missing the `values.yaml` entries for the global parameters injected by Argo CD. This will cause test failures from the Helm linter.

athornton commented:
Did some more linting.  I don't understand why the linter is failing on strimzi-request-operator but it looks like it doesn't know it's supposed to template it?  Maybe because it's not in a "templates" directory outside the symlink?  I'll ask Angelo tomorrow.

athornton commented:
Well, argocd is gonna need some work.  Trying to DRY it and make it use global secret path is hanging in app installation.  So...let's punt that till later.  This work does quite a bit of cleanup for our other services, anyway.

rra commented:
The strimzi-registry-operator chart isn't valid if I'm reading this correctly. It needs the normal stuff like a `Chart.yaml` and a `values.yaml`.

I'm confused by why there's still a separate `services/strimzi-registry-operator` chart. The users of the strimzi-registry-operator chart should have a symlink in their `charts` subdirectory that points to `charts/strimzi-registry-operator`.

athornton commented:
Now *I* am confused.  I haven't deployed strimzi-registry-operator, apparently.  I thought the minikube deployment was doing so since it has values-minikube.yaml but it's got enabled=false in science-platform.

However, when I run
```
adam@m1-wired:~/git/phalanx/services/strimzi-registry-operator$ helm template -f values.yaml -f values-minikube.yaml .
```
I get a warning, but that warning is accurate, but it produces what looks like a reasonable manifest.
```
walk.go:74: found symbolic link in path: /Users/adam/git/phalanx/services/strimzi-registry-operator/templates resolves to /Users/adam/git/phalanx/charts/strimzi-registry-operator/templates
```

athornton commented:
OK....strimzi-registry-operator is back to master.  I could try changing the repository to point to the local charts directory (for that and sasquatch) if we want, or we could this good enough.

rra commented on services/datalinker/templates/ingress.yaml:
```diff
@@ -0,0 +1,30 @@
+      http:
+        paths:
+          - path: "/api/datalink"
+            pathType: ImplementationSpecific
```
`pathType` should be `Prefix`

athornton replied:
Done.

rra commented on services/obstap/Chart.yaml:
```diff
@@ -1,10 +1,6 @@
-- name: pull-secret
-  version: 0.1.2
-  repository: https://lsst-sqre.github.io/charts/
+apiVersion: v1
```
`apiVersion` should be `v2`

athornton replied:
Done.

rra commented on services/obstap/templates/tap-ingress-anonymous.yaml:
```diff
@@ -0,0 +1,45 @@
+    - host: {{ required "global.host must be set" .Values.global.host | quote }}
+      http:
+        paths:
+          - path: "/api/obstap/(availability)"
+            pathType: "ImplementationSpecific"
+            backend:
+              service:
+                name: {{ template "cadc-tap-postgres.fullname" . }}
+                port:
+                  number: 80
+          - path: "/api/obstap/(capabilities)"
+            pathType: "ImplementationSpecific"
+            backend:
+              service:
+                name: {{ template "cadc-tap-postgres.fullname" . }}
+                port:
+                  number: 80
+          - path: "/api/obstap/(swagger-ui.*)"
+            pathType: "ImplementationSpecific"
+            backend:
+              service:
+                name: {{ template "cadc-tap-postgres.fullname" . }}
+                port:
+                  number: 80
```
This is already a regex, so these can all be merged into one rule, such as for the tap service.

athornton replied:
Done.

rra commented on services/plot-navigator/values.yaml:
```diff
@@ -0,0 +1,26 @@
+image:
+  repository: lsstdm/pipetask-plot-navigator
+  tag: ""
+
+# Environment variables to be passed to panel.
+# e.g. butler configuration and auth params.
+environment: {}
+
+ingress:
+  gafaelfawrAuthQuery: "scope=exec:portal&delegate_to=plotnavigator"
+  annotations: {}
+
```
Need helm-docs comments

athornton replied:
Helm docs added.  Charts deleted.

rra reviewed (commented):
There are still files in `charts/strimzi-registry-operator`. Presumably those should now be deleted?

I've resolved the inline comments that were obsolete. The remaining ones from the previous review all still need to be addressed.

athornton commented:
Gaaah.  `git status` might have been helpful to me.

athornton commented on science-platform/templates/postgres-application.yaml:
```diff
@@ -21,6 +21,14 @@ spec:
     targetRevision: {{ .Values.revision }}
     helm:
+      parameters:
+        - name: "global.host"
+          value: {{ .Values.fqdn | quote }}
+        - name: "global.baseUrl"
+          value: "https://{{ .Values.fqdn }}"
```
Done.

athornton commented on services/cachemachine/templates/serviceaccount.yaml:
```diff
@@ -8,6 +8,8 @@ metadata:
   annotations:
     {{- toYaml . | nindent 4 }}
   {{- end }}
+imagePullSecrets:
+  - name: "pull-secret"
```
Done

athornton commented:
I think this time, maybe, I actually HAVE addressed all the review comments.  At this point the only thing I know of that is still outstanding is:

global.vaultSecretsPathPrefix or global.vaultSecretsPath ?

I prefer the shorter version, but I don't really care.  @rra and @jonathansick, please pick a winner between the two of those, and I'll go make it consistent.
